### ISSUE TITLE
Walk HDF5 data tree? 

### ISSUE THREAD

**Issue by bsrdjan** [HDF-NI/hdf5.node#51]
Any example how to walk HDF5 data tree using hdf5.node?

Following [HDF5 examples](https://support.hdfgroup.org/HDF5/Tutor/discover.html#trav), `visit` and `iterate` methods shall be used, like done in h5py for example: https://github.com/h5py/h5py/issues/406.

**Comment by rimmartin** [comment]
Good Idea.  I'll add the technique.  Will commit probably on the weekend with some hdf5 1.10.x changes

**Comment by rimmartin** [comment]
Added an iterate on files and groups.
https://github.com/HDF-NI/hdf5.node/blob/master/test/test_h5lt.js#L840
more will be added.

**Comment by rimmartin** [comment]
added recursive visiting on links. Returns an "xpath" of the groups from the root group or the file itself.

looking at the object visiting next
